src/cswinrt/code_writers.h

```diff
@@ -5905,15 +5961,21 @@ IntPtr instance = _factory.ActivateInstance();
 return ObjectReference<IInspectable.Vftbl>.Attach(ref instance).As<I>();
 }
 
+public void RunClassConstructor()
```

Comment:
I don't see any tests/samples that exercise this.  But could we do away with IManagedActivationFactory (and the breaking change) by just calling RunClassConstructor from a static constructor of the activation factory?

Reply:
The WinUI test triggers it.  In the below code, when you set custom properties in a XAML file, XAML QIs for IManagedActivationFactory and does a call to RunClassConstructor.

```       <custom:CustomButton x:Name="myButton" Click="myButton_Click" OverrideEntered="False">Custom</custom:CustomButton>```

I will try removing the interface and calling it as part of ActivationFactory and see what happens, but WinUI may break I think.

Reply:
Looks like WinUI calls it [here](https://dev.azure.com/microsoft/WinUI/_git/microsoft-ui-xaml-lift?path=%2Fdxaml%2Fxcp%2Fcore%2Fmetadata%2FReflectionAPI.cpp&version=GBmaster&line=467&lineEnd=468&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents).  If the activation factory type is accessed (causing the static ctor to execute), and the cast to IManagedActivationFactory then fails, that might still be a reliable mechanism.

Reply:
Went ahead and removed it as we don't see a disadvantage or scenario that might be impacted by doing that and allows us to not increment assembly version.